json_serialization/writer.nim

```diff
@@ -124,9 +124,16 @@ proc writeIterable*(w: var JsonWriter, collection: auto) =
 
   append ']'
 
-proc writeArray[T](w: var JsonWriter, elements: openarray[T]) =
+proc writeArray*[T](w: var JsonWriter, elements: openarray[T]) =
   writeIterable(w, elements)
 
+# this construct catches `array[N, char]` which otherwise won't decompose into
+# openArray[char] - we treat any array-like thing-of-characters as a string in
+# the output
+template isStringLike(v: string|cstring|openArray[char]|seq[char]): bool = true
+template isStringLike[N](v: array[N, char]): bool = true
```

Comment:
The `array` and `seq` cases could be a bit surprising perhaps, but we could give it a try.
If we support them here, we'll should reflect that in the reader as well. You'll be then able to use the [`roundtripTest`](https://github.com/status-im/nim-json-serialization/blob/3f9021a139394cd438dd20f2e6cf13034bfd67bc/tests/test_serialization.nim#L139) API to write a good test.

Reply:
right now `char` leads to a compile error so we're adding to existing behaviour

Reply:
array is needed for the below test - the type is transported perfectly, and not as `openArray[char]` - but if `seq[char]` converst to `openarray[char]` it would be strange to have different behaviours.